flink-runtime/src/main/java/org/apache/flink/runtime/blob/BlobUtils.java

```diff
@@ -227,7 +227,7 @@ static File getStorageLocation(File storageDir, JobID jobID, String key) {
 	private static File getJobDirectory(File storageDir, JobID jobID) {
 		final File jobDirectory = new File(storageDir, JOB_DIR_PREFIX + jobID.toString());
 
-		if (!jobDirectory.exists() && !jobDirectory.mkdirs()) {
```

Comment:
the new way is thread-safe - imagine a directory being concurrently created after another thread has failed the `exists()` part - not sure we actually fix a bug nowadays since the use may be guarded...so this is probably more of a "just-in-case" thing or if usage patterns change